airflow/utils/task_group.py

```diff
@@ -344,3 +357,52 @@ def get_current_task_group(cls, dag: Optional["DAG"]) -> Optional[TaskGroup]:
                 return dag.task_group
 
         return cls._context_managed_task_group
+
+
+T = TypeVar("T", bound=Callable)  # pylint: disable=invalid-name
+
+
+def taskgroup(python_callable: Optional[Callable] = None, *tg_args, **tg_kwargs) -> Callable[[T], T]:
+    """
+    Python TaskGroup decorator. Wraps a function into an Airflow TaskGroup.
+    Accepts kwargs for operator TaskGroup. Can be used to parametrize TaskGroup.
+
+    :param python_callable: Function to decorate
+    :param tg_args: Arguments for TaskGroup object
+    :type tg_args: list
+    :param tg_kwargs: Kwargs for TaskGroup object.
+    :type tg_kwargs: dict
+    """
+
+    def wrapper(f: T):
+        # Setting group_id as function name if not given
+        if len(tg_args) == 0 and 'group_id' not in tg_kwargs.keys():
+            tg_kwargs['group_id'] = f.__name__
```

Comment:
Can we add logic to autogenerate group_id if it already exists on the DAG? 

A taskgroup decorated function can be invoked multiple times, this should generate multiple taskgroups. 

We can follow a similar approach than @task decorator where we append `__1` to the end. Probs we can reuse the same regex.

Reply:
Sure, can you guide me how to get list of existing group_id inside taskgroup decorator?

Reply:
I have added auto generate group_id if it already exists  same as task. and also added test case for it.